cmake: Add warning when unit testing is disabled due to assertions being disabled. (#5204)
CVC4/CVC4: INSTALL.md

- **api tests** in directory `test/api`
  (label: **api**)
- **unit tests** in directory `test/unit`
  (label: **unit**)

### Testing System Tests

The system tests are not built by default.

    make apitests                         # build and run all system tests
    make <api_test>                       # build test/system/<system_test>.<ext>
    ctest api/<api_test>                  # run test/system/<system_test>.<ext>

All system test binaries are built into `<build_dir>/bin/test/system`.

We use prefix `api/` + `<api_test>` (for `<api_test>` in `test/api`)
as test target name.  

    make ouroborous                       # build test/api/ouroborous.cpp
    ctest -R ouroborous                   # run all tests that match '*ouroborous*'
                                          # > runs api/ouroborous
    ctest -R ouroborous$                  # run all tests that match '*ouroborous'
                                          # > runs api/ouroborous
    ctest -R api/ouroborous$              # run all tests that match '*api/ouroborous'
                                          # > runs api/ouroborous
### Testing Unit Tests

The unit tests are not built by default.

> To configure CVC4 with unit tests, assertions must be enabled, and building a
> non-static binary must be configured.

Reviewer: ```suggestion
Note that CVC4 can only be configured with unit tests in non-static builds with
assertions enabled.
```
